src/main/scala/org/apache/spark/sql/pulsar/PulsarMetadataReader.scala

```diff
@@ -59,31 +60,39 @@ private[pulsar] case class PulsarMetadataReader(
 
   def setupCursor(startingPos: PerTopicOffset): Unit = {
     startingPos match {
-      case off: SpecificPulsarOffset => setupCursorByMid(off)
-      case time: SpecificPulsarStartingTime => setupCursorByTime(time)
+      case off: SpecificPulsarOffset => setupCursorByMid(off, predefinedSubscription)
+      case time: SpecificPulsarStartingTime => setupCursorByTime(time, predefinedSubscription)
       case s => throw new UnsupportedOperationException(s"$s shouldn't appear here, a bug occurs.")
     }
   }
 
-  def setupCursorByMid(offset: SpecificPulsarOffset): Unit = {
+  def setupCursorByMid(offset: SpecificPulsarOffset, subscription: Option[String]): Unit = {
     offset.topicOffsets.foreach {
```

Comment:
I assume there are multiple topics involved, and if we use one subscription name for all the different topics, wondering what would be the case when all the topics have the same subscription name

Reply:
Topics have same name subscriptions should be fine.

We can iterate on this issue based on user feedback